source/solvers/vof_assemblers.cc

```diff
@@ -195,19 +195,35 @@ VOFAssemblerCore<dim>::assemble_rhs(VOFScratchData<dim> &      scratch_data,
       // Store JxW in local variable for faster access;
       const double JxW = JxW_vec[q];
 
-      // Shock capturing viscosity term
-      const double order = scratch_data.fe_values_vof.get_fe().degree;
+      // Implementation of a DCDD shock capturing scheme.
+      // For more information see
+      // Tezduyar, T. E., & Park, Y. J. (1986). Discontinuity-capturing
+      // finite element formulations for nonlinear
+      // convection-diffusion-reaction equations. Computer methods in
+      // applied mechanics and engineering, 59(3), 307-325.
+
+      // Gather the order of the VOF interpolation
+      const double order = this->fem_parameters.VOF_order;
 
+      // Calculate the artificial viscosity of the shock capture
       const double vdcdd = (0.5 * h) * (velocity.norm() * velocity.norm()) *
                            pow(phase_gradient.norm() * h, order);
 
-      const double   tolerance = 1e-12;
-      Tensor<1, dim> s         = velocity / (velocity.norm() + tolerance);
+      const double tolerance = 1e-12;
+
+      // We neglect to remove the diffusion aligned with the velocity
+      // as is done in the original article. We re-enable those
+      // terms if artificial diffusion becomes a problem
+      // Tensor<1, dim> s = velocity / (velocity.norm() + 1e-12);
+      // const Tensor<2, dim> k_corr      = (r * s) * outer_product(s,
+      // s);
```

Comment:
Is this re-enabled frequently? If yes, isn't it better to use an if that allows to enable it easily?

Reply:
This shouldnt be re-enabled. :).